lntest/harness.go

```diff
@@ -552,7 +570,43 @@ func (n *NetworkHarness) RestartNode(node *HarnessNode, callback func() error) e
 		}
 	}
 
-	return node.start(n.lndErrorChan)
+	if err := node.start(n.lndErrorChan); err != nil{
+		return err
+	}
+
+	return ensureServerStarted(node)
+}
+
+// ensureServerStarted  verifies that the lnd server started.
+// It is done by using RPC calls to DisconnectPeerRequest which returns an error "chain backend is still syncing"
+// before the server started.
+// TODO (Offer): replace this hack with a field in getInfo that indicates that the server staretd
+
+func ensureServerStarted (node *HarnessNode) error{
```

Comment:
If `synced_with_chain` is true within the `getinfo` response, then we can most likely guarantee that the server has started.